src/otaclient_common/linux.py

```diff
@@ -233,29 +201,62 @@ def pyinstaller_aware_subprocess_run(args, *p_pargs, env=None, **p_kwargs):
     return subprocess.run(args, *p_pargs, env=_parsed_env, **p_kwargs)
 
 
+@overload
 def subprocess_run_wrapper(
     cmd: str | list[str],
     *,
     check: bool,
     check_output: bool,
-    chroot: StrOrPath | None = None,
+    chroot: StrOrPath,
     set_host_mnt_ns: bool = False,
     env: Optional[dict[str, str]] = None,
     timeout: Optional[float] = None,
+) -> subprocess.CompletedProcess[bytes]: ...
+
+
+@overload
+def subprocess_run_wrapper(
+    cmd: str | list[str],
+    *,
+    check: bool,
+    check_output: bool,
+    chroot: None = None,
+    set_host_mnt_ns: None = None,
+    env: Optional[dict[str, str]] = None,
+    timeout: Optional[float] = None,
+) -> subprocess.CompletedProcess[bytes]: ...
+
+
+def subprocess_run_wrapper(
+    cmd: str | list[str],
+    *,
+    check: bool,
+    check_output: bool,
+    chroot: StrOrPath | None = None,
+    set_host_mnt_ns: bool | None = None,
+    env: Optional[dict[str, str]] = None,
+    timeout: Optional[float] = None,
```

Comment:
Using precise typing to ensure `chroot` is set when `set_host_mnt_ns` is not False. As `subprocess_run_wrapper` is used internally within otaclient, using typing is totally  enough.